base/file.jl

```diff
@@ -230,23 +230,27 @@ julia> mkpath("intermediate_dir/actually_a_directory.txt") # creates two directo
 julia> isdir("intermediate_dir/actually_a_directory.txt")
 true
 
+julia> mkpath("my/test/dir/") # returns the original `path`
```

Comment:
This extra comment makes sense in the context of this PR but if someone read it in the future one would wonder why this is here (after all this is just the documented behavior). The future reader would not know that there used to be a bug here

Reply:
It is used to clarify the behavior, but I am ok to remove it if we believe it is unnecessary or harmful.